src/Provider/Google.php

```diff
@@ -19,7 +19,8 @@ class Google extends AbstractProvider
     protected $accessType;
 
     /**
-     * @var string If set, this will be sent to google as the "hd" parameter.
+     * @var string Comma-separated list of domains or domain regular expressions.
+     *             If only one regular value is passed, it will be sent to google as the "hd" parameter.
```

Comment:
Where is documented on Google's Developer website?

Reply:
It's the documentation of the function prototype rather than the Google API.
`If only one regular value is passed` ... **to this function,** then `it will be sent to google as the "hd" parameter`.

Citing [documentation](https://developers.google.com/identity/protocols/OpenIDConnect#hd-param):

> The hd (hosted domain) parameter streamlines the login process for G Suite hosted accounts. By including the domain of the G Suite user (for example, mycollege.edu), you can indicate that the account selection UI should be optimized for accounts at that domain. To optimize for G Suite accounts generally instead of just one domain, use an asterisk: hd=*.
> 
> Don't rely on this UI optimization to control who can access your app, as client-side requests can be modified. Be sure to validate that the returned ID token has an hd claim value that matches what you expect (e.g. mycolledge.edu). Unlike the request parameter, the ID token claim is contained within a security token from Google, so the value can be trusted.
> 

If someone want to allow access from `mycompany.(org|com),mysubsidiary.tel`:
* `hd` Google parameter can not contain multiple values or regexp. Only one regular GSuite domain.
* `hd=mycompany.org` would block `mycompany.org`

What we do want is to  **store** the `hd` value (in order to do the final access checks) **but** we do **not** pass any `hd` parameter to GSuite connection screen because this screen would then block the access all other client domains.

Reply:
That seems like an implementation detail that should be handled by the end user, rather than this package. The scenario you are describing would benefit from `hd=*` and then the usage of this package would check the resulting claim for a match.

From my standpoint, this PR makes the `hostedDomain` less obvious and more confusing to use, because it no longer aligns with the Google Developer documentation.

Reply:
- PR is fully backward compatible: Existing use of the `hd` parameter are left unaffected, completely supported and working as expected.
- Extra-power is granted to those who opt for regexp/domain-list but it's still fine for package's consumers to ignore it. (= this enhancement does not make noise)
- For configuration using multiple domain, it'd not be a surprise nor confusing to see free-form domain-name on Google login page
- Overall added complexity is 40 lines. IMHO having that facility in the package is bonus that outweigh the complexity cost.
- An app' restricting Google OIDC login for a set of domains would likely have to implement 95% of this very PR. (I already had to do such implementation myself for two different CMS in the last 6 months). It means:
  - Implementing `isHostedDomainMultiple` & co
  - Then, if true, passing `hd=*` to `oauth2-google`
  - Then checking auth'ed user-domain value returned by `AbstractProvider::getResourceOwner()`.
  - **But** the `getResourceOwner()` is generic and it could be a burden for implementers to treat Google domain-whitelisting specifically... in a part of there code which is probably provider-independent.
- It could be frustrating a simple-yet-powerful way to restrict GSuite-domain could not be available in this library (it's the point of a library after all), while everything is in place here to make it more powerful at a small price.

Going further about use-cases: Google OIDC can be used for
1. public subscription website (no `hd`)
2. internal authentication (probably the more common case).

In the later case, it's likely to be the *only* provider enabled. And I think _OpenId logins plugins/modules (WP/Drupal/Grav/...)_ would prefer to have `hd` domain restriction handled out of box by `oauth2-google/src/Provider/Google.php` rather than having to pass `hd=*` (simply to please the API), then writing semi-generic restriction based on (generic) `AbstractProvider::getResourceOwner()`.

Reply:
Hi,
Did you considered above arguments / reconsidered point of view?